src/android/wifiwizard2/WifiWizard2.java

```diff
@@ -1198,31 +1231,80 @@ private boolean isWifiEnabled(CallbackContext callbackContext) {
    * This method takes a given String, searches the current list of configured WiFi networks, and
    * returns the networkId for the network if the SSID matches. If not, it returns -1.
    */
-  private int ssidToNetworkId(String ssid) {
-
+  private int ssidToNetworkId(String ssid, String authType) {
     try {
 
       int maybeNetId = Integer.parseInt(ssid);
       Log.d(TAG, "ssidToNetworkId passed SSID is integer, probably a Network ID: " + ssid);
       return maybeNetId;
 
     } catch (NumberFormatException e) {
-
       List<WifiConfiguration> currentNetworks = wifiManager.getConfiguredNetworks();
       int networkId = -1;
-
-      // For each network in the list, compare the SSID with the given one
+      // For each network in the list, compare the SSID with the given one and check if authType matches
       for (WifiConfiguration test : currentNetworks) {
-        if (test.SSID != null && test.SSID.equals(ssid)) {
-          networkId = test.networkId;
+        if (test.SSID != null) {
```

Comment:
Suggest changing `test` to something more appropriate.

Reply:
Pretty trivial change but I don't like variables `test` in production code.

Reply:
Wow forgot this PR was open ^^currently messing with master to get IP on iOS. will refactor for better PR later

Reply:
All good - Did you ever get around to fixing it up?